analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/CryptographyRuleSymbolicCheck.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Security.Cryptography;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
+
+public abstract class CryptographyRuleSymbolicCheck : SymbolicRuleCheck
+{
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var state = context.State;
```

Comment:
This varable doens't make sense. `state` doesn't evolve it's value over the course of this method. And `else` returns `context.State` anyway
```suggestion
```

Reply:
The goal here was to repeat something with fewer chars rather than `Context.State` all the time (last if was just forgotten).
So the idea is that we do this only if there's some evolution of the variable value?

Reply:
When you introduce `state`, you introduce possible confusion between `state` and `context.State` - and it actually happened below, because both got used.

So in this case, it's not worth it. I'd declare `state` only when it evolves over the course of the method and consequent code needs the updated version. Example:
https://github.com/SonarSource/sonar-dotnet/blob/84dac82ba11def0db40f9387c928df4605b51b1f/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs#L37-L52